### PR TITLE
E2E-634: expose rar requirement id on appointment apis

### PR DESCRIPTION
**Pull request by axle-h**
Appointment consumers are interested in whether or not an appointment relates to a RAR requirement. This compliments the RAR_ACTIVITY flag for determining whether the appoint had anything to do with RAR. Decided not to include the entire requirement in the response for performance.

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/transformers/AppointmentTransformer.java [hunk 1]
```diff
@@ -48,6 +51,15 @@ public static AppointmentDetail appointmentDetailOf(Contact contact) {
             .sensitive(ynToBoolean(contact.getSensitive()))
             .outcome(appointmentOutcomeOf(contact))
             .rarActivity(ynToBoolean(contact.getRarActivity()))
+            .rarRequirementId(
+                Optional.ofNullable(contact.getNsi())
+                    .map(Nsi::getRqmnt)
```
Comment (mahalrmoj) [inline src/main/java/uk/gov/justice/digital/delius/transformers/AppointmentTransformer.java:56]: Do you need to filter out inactive and soft deleted requirements as both features seem to be used. They were a problem for R&M but may not be for your context.

File: src/main/java/uk/gov/justice/digital/delius/jpa/standard/entity/RequirementTypeMainCategory.java [hunk 2]
```diff
@@ -33,4 +33,8 @@
 
     @Column(name="RESTRICTIVE")
     private String restrictive;
+
+    public boolean isRarRequirementType() {
+        return code != null && code.equals(REHABILITATION_ACTIVITY_REQUIREMENT_CODE);
+    }
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/jpa/standard/entity/RequirementTypeMainCategory.java:39]: very minor style thing I guess you could have written `return REHABILITATION_ACTIVITY_REQUIREMENT_CODE.equals(code);`

**Review by andymarke — APPROVED** [review]
LGTM

**Review by andymarke — DISMISSED** [review]
LGTM

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**